src/app/content/utils/processBrowserRedirect.ts

```diff
@@ -2,21 +2,28 @@ import { History } from 'history';
 import { Redirects } from '../../../../data/redirects/types';
 import { RouterService } from '../../navigation/routerService';
 import { assertWindow } from '../../utils';
-import { matchForRoute } from '../../navigation/utils';
-import { external } from '../../errors/routes';
+import { Dispatch } from '../../types';
+import { replace } from '../../navigation/actions';
+import { AnyMatch } from '../../navigation/types';
 
-export const processBrowserRedirect = async(services: {router: RouterService, history: History}) => {
+export const processBrowserRedirect = async(services: {
+  router: RouterService,
+  history: History,
+  dispatch: Dispatch
+}) => {
   const window = assertWindow();
+  const base = window.location.origin;
+
   const redirects: Redirects = await fetch('/rex/redirects.json')
     .then((res) => res.json())
     .catch(() => []);
 
   for (const {from, to} of redirects) {
     if (from === services.history.location.pathname) {
-      if (matchForRoute(external, services.router.findRoute(to))) {
-        window.location.href = window.location.origin + to;
-      }
-      services.history.replace(to);
+      const {pathname, search, hash} = new URL(to, base);
+      const match = services.router.findRoute({pathname, search, hash, state: {}});
+      services.history.replace(to); // hack to show blank page instead of flashing rex when target is external
```

Comment:
i believe if you delete [this return statement](https://github.com/openstax/rex-web/blob/main/src/app/navigation/middleware.ts#L27C7-L27C13) then it'll route to the error route screen, which renders a loader, that might be a better solution than this to prevent displaying rex

Reply:
the `?message=retired` redirects I'm trying to handle don't hit that return statement because they don't match a notFound or external content route